### ISSUE TITLE
Sabbaticals

### ISSUE THREAD

**Issue by nayafia** [pybee/paying-the-piper#28]
I'm inspired by the GSOC model of funding student developers to work on open source projects.

The part I think they get right is A) making it a dedicated, focused period of time vs. ad-hoc "whenever I get around to it", and B) providing stipends to make it financially feasible.

The part I think could be iterated upon is A) providing stipends to experienced contributors, not just students, and B) widening the scope of projects to include more niche stuff. And, obviously, C) making it a year-round thing.

What do you think about providing paid "sabbaticals" for contributors to work on an open source project for a dedicated 1-3 months, a la GSOC? It would be harder for those with full-time jobs, I think, but people might be able to knock out a lot instead of dragging their to-dos out with a couple hours/wk here and there.

**Comment by seiyria** [comment]
This would be nice, but as you said - it isn't really feasible for those working full time, nor is it feasible for those that can't fund it. This really boils down to a "where does the cashflow come from" question, I think.

**Comment by nayafia** [comment]
Yeah, where the $ comes from (and how much there is available) really sets the parameters of what's feasible. I think companies are a good place to start. Stripe started an experiment last year called the Open-Source Retreat where they provided $20K grants for 2-4 contributors to work on their projects for 3 months.

**Comment by freakboy3742** [comment]
This is effectively what #3 does - Kickstarter-style projects lend themselves to raising "1-3 months" of money to work on a feature. However, as @seiyria points out, it doesn't solve the long term maintenance problem.

**Comment by nayafia** [comment]
I think sabbaticals could work for maintenance as well, not just new features. Hard part is having to fit the work into cycles: say, 3 months of paid time to knock out the big to-dos, enough to sustain another 9 months of ad hoc work.

**Comment by Hnasar** [comment]
This model has also been adopted by http://outreachy.org/ to steady success. The funding comes from donations from companies interested in funding diversity outreach.
